hibernate-core/src/main/java/org/hibernate/engine/query/spi/EntityGraphQueryHint.java

```diff
@@ -37,30 +39,32 @@
  *
  * @author Brett Meyer
  */
-public class EntityGraphQueryHint {
-	private final RootGraph<?> graph;
+public class EntityGraphQueryHint implements AppliedGraph {
```

Comment:
I'm confused why you say that the other HHH-12713 PR needs to be applied first.  Unless I miss something, this PR already includes those changes.  Here is a good example.  This exact change exists in that other PR - https://github.com/hibernate/hibernate-orm/pull/2604/files#diff-42aef08c22c86f0e459a475e822240a4R42

Reply:
Yes. You can't send a PR "B" based on another PR "A" without including the commits from PR "B".

I said the other PR needs to be merged first, because we need to review that code first. If we were to merge my PR now, we would effectively merge yours too.